GeeksCoreLibrary/Modules/Payments/Services/BuckarooService.cs

```diff
@@ -209,17 +223,36 @@ public async Task<StatusUpdateResult> ProcessStatusUpdateAsync(OrderProcessSetti
             }
 
             var buckarooSettings = (BuckarooSettingsModel)paymentMethodSettings.PaymentServiceProvider;
-            var buckarooClient = new BuckarooSdk.SdkClient();
 
             // Read the entire body, which should be a JSON body from Buckaroo.
             using var reader = new StreamReader(httpContextAccessor.HttpContext.Request.Body);
             var bodyJson = await reader.ReadToEndAsync();
+
+            if (!String.IsNullOrEmpty(bodyJson))
+            {
+                return await HandleJsonStatusUpdate(buckarooSettings, invoiceNumber, bodyJson);
+            }
+
+            // No JSON body found, so we'll try to handle it as a form.
+            return await HandleFormStatusUpdate(buckarooSettings, invoiceNumber);
```

Comment:
Gaat dit wel goed? Volgens mij is de request body altijd gevuld, ongeacht wat het content-type is. Je gaat er nu vanuit dat als die gevuld is, dat het dan JSON is, maar dat zou ook gevuld moeten zijn als het form-data is.

Reply:
Ik heb dit getest, en kan bevestigen dat de request body leeg is als form values worden meegegeven. Wellicht is het beter om dit via een instelling te beheren, dus dat je bij de Buckaroo instellingen kan instellen wat de push methode is van Buckaroo. Die kan op "httppost", "json" of "soap" staan (al denk ik dat we "soap" niet moeten gaan ondersteunen, of pas wanneer dit echt nodig gaat zijn). Kijken naar content type is natuurlijk ook een optie.

Dit geldt ook voor de signature validatie, die staat nu op SHA1, terwijl Buckaroo ook SHA256 en SHA512 ondersteunt (en BCRYPT7, BCRYPT10 en WHIRLPOOL, maar die zeggen me niks).

Reply:
Ik heb dit nu uitgebreid zodat er wordt gekeken naar de ingestelde push content type en HASH-methode om te bepalen welke status update methode gebruikt moet worden, en hoe de signature wordt gehashed.